packages/code-studio/src/styleguide/StyleGuide.tsx

```diff
@@ -71,24 +76,34 @@ function StyleGuide(): React.ReactElement {
 
         <Flex
           {...stickyProps}
-          UNSAFE_className={HIDE_FROM_E2E_TESTS_CLASS}
+          // UNSAFE_className={HIDE_FROM_E2E_TESTS_CLASS}
           marginTop={-56}
           top={20}
           gap={10}
           alignItems="end"
         >
           {hasMultipleThemes ? <ThemePicker /> : null}
           <SamplesMenu />
+          {isTestMode && (
+            <input
```

Comment:
Since we already use hash routing to identify which section is selected, we could probably simplify this by listening to the hash change event instead of requiring this input.

```typescript
 useEffect(() => {
    const hashChangeHandler = () => {
      setTargetSection(window.location.hash.replace('#', ''));
    };

    window.addEventListener('hashchange', hashChangeHandler);

    return () => {
      window.removeEventListener('hashchange', hashChangeHandler);
    };
  }, []);
  ```
  I think the unit test could then just set the `window.location.hash` and it "should" just work. Setting the hash alone shouldn't reload the page